src/js/components/aboutTheData/AgencyDetailsPage.jsx

```diff
@@ -41,20 +42,30 @@ const AgencyDetailsPage = () => {
     const closeModal = () => setShowModal('');
 
     useEffect(() => {
-        // TODO - replace mock API request
-        const agencyOverviewRequest = mockFetchAgencyOverview(agencyCode);
-        agencyOverviewRequest.promise
-            .then((res) => {
+        const getOverviewData = async () => {
```

Comment:
I'm curious how we might be able to establish this pattern via a custom hook...... not sure